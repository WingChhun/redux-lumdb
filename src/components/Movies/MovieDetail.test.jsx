import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import MovieDetail from "./MovieDetail.jsx";

Enzyme.configure({adapter: new Adapter()})

describe("MovieDetail", () => {

    const props = {

        getMovie: jest.fn(),
        resetMovie: jest.fn(),
        movie: {
            id: 1234,
            title: "Hello",
            poster_path: "hello world",
            backdrop_path: "/dude",
            release_date: "yo",
            overview: "Boy meets world"

        },
        backdrop: "/backdrop",
        match: {
            params: {
                id: 1234
            }
        }

    }

    it("Renders properly if given movies", () => {
        const wrapper = shallow(<MovieDetail {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it("componentWillReceiveProps shoul;d update movies State", () => {

        const newProps = {
            ...props,
            movie: {
                id: 1234,
                title: "This should work"
            }
        }
        const wrapper = shallow(<MovieDetail {...props}/>);
        const instance = wrapper.instance();

        instance.componentWillReceiveProps(newProps);

        expect(instance.state.movie).toBe(newProps.movie);

    });

    it("componentWillMount() will call getMovie", () => {

        const wrapper = shallow(<MovieDetail {...props}/>);
        const instance = wrapper.instance();

        instance.componentWillMount();

        expect(props.getMovie).toHaveBeenCalled()

    });

    it("componentWIllUnmount() will call resetMovie", () => {

        const wrapper = shallow(<MovieDetail {...props}/>);
        const instance = wrapper.instance();

        instance.componentWillUnmount();

        expect(props.resetMovie).toHaveBeenCalled();

    });
});