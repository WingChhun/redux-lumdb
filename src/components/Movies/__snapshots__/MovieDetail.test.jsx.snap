// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail Renders properly if given movies 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieDetail
    backdrop="/backdrop"
    getMovie={[Function]}
    match={
        Object {
            "params": Object {
              "id": 1234,
            },
          }
    }
    movie={
        Object {
            "backdrop_path": "/dude",
            "id": 1234,
            "overview": "Boy meets world",
            "poster_path": "hello world",
            "release_date": "yo",
            "title": "Hello",
          }
    }
    resetMovie={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backdrop": "http://image.tmdb.org/t/p/w1280/dude",
      "children": <styled.div>
        <e
                duration={200}
                element="div"
                id="1234"
        >
                <styled.img
                        alt="Hello"
                        src="http://image.tmdb.org/t/p/w154hello world"
                />
        </e>
        <div>
                <h1>
                        Hello
                </h1>
                <h1>
                        Hello
                </h1>
                <h3>
                        yo
                </h3>
                <p>
                        Boy meets world
                </p>
        </div>
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <e
            duration={200}
            element="div"
            id="1234"
>
            <styled.img
                        alt="Hello"
                        src="http://image.tmdb.org/t/p/w154hello world"
            />
</e>,
          <div>
            <h1>
                        Hello
            </h1>
            <h1>
                        Hello
            </h1>
            <h3>
                        yo
            </h3>
            <p>
                        Boy meets world
            </p>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.img
              alt="Hello"
              src="http://image.tmdb.org/t/p/w154hello world"
/>,
            "duration": 200,
            "element": "div",
            "id": "1234",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Hello",
              "src": "http://image.tmdb.org/t/p/w154hello world",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Hello
</h1>,
              <h1>
                Hello
</h1>,
              <h3>
                yo
</h3>,
              <p>
                Boy meets world
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello",
              },
              "ref": null,
              "rendered": "Hello",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello",
              },
              "ref": null,
              "rendered": "Hello",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "yo",
              },
              "ref": null,
              "rendered": "yo",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Boy meets world",
              },
              "ref": null,
              "rendered": "Boy meets world",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backdrop": "http://image.tmdb.org/t/p/w1280/dude",
        "children": <styled.div>
          <e
                    duration={200}
                    element="div"
                    id="1234"
          >
                    <styled.img
                              alt="Hello"
                              src="http://image.tmdb.org/t/p/w154hello world"
                    />
          </e>
          <div>
                    <h1>
                              Hello
                    </h1>
                    <h1>
                              Hello
                    </h1>
                    <h3>
                              yo
                    </h3>
                    <p>
                              Boy meets world
                    </p>
          </div>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <e
              duration={200}
              element="div"
              id="1234"
>
              <styled.img
                            alt="Hello"
                            src="http://image.tmdb.org/t/p/w154hello world"
              />
</e>,
            <div>
              <h1>
                            Hello
              </h1>
              <h1>
                            Hello
              </h1>
              <h3>
                            yo
              </h3>
              <p>
                            Boy meets world
              </p>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.img
                alt="Hello"
                src="http://image.tmdb.org/t/p/w154hello world"
/>,
              "duration": 200,
              "element": "div",
              "id": "1234",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Hello",
                "src": "http://image.tmdb.org/t/p/w154hello world",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Hello
</h1>,
                <h1>
                  Hello
</h1>,
                <h3>
                  yo
</h3>,
                <p>
                  Boy meets world
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hello",
                },
                "ref": null,
                "rendered": "Hello",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hello",
                },
                "ref": null,
                "rendered": "Hello",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "yo",
                },
                "ref": null,
                "rendered": "yo",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Boy meets world",
                },
                "ref": null,
                "rendered": "Boy meets world",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
